docs/DeveloperGuide.md

```diff
@@ -4,35 +4,261 @@
+- `Sales` is instantiated with an ArrayList of `OrderList`, forming a one-to-many relationship with `OrderList`.
+- Lastly, the `CurrentDate` class keeps track of the current operating day of the cafe.
+
+
+## Features
+
+### Add Dish
+
+![Add Dish Execution](images/sequence/AddDishCommand_execute.png)
+*Figure 8: Execution of add_dish command*
+
+API: [AddDishCommand.java](https://github.com/AY2324S1-CS2113-T17-2/tp/blob/master/src/main/java/seedu/cafectrl/command/AddDishCommand.java)
+
+The `add_dish` command, add a dish to the `Menu` object and prints out a formatted message to state the name, price and ingredients entered for the dish.
+
+when the `execute()` method from `AddDishCommand` is called in the main class `CafeCtrl`, the `addDish()` method is first called to add the `Dish` object to the `Menu`. It will then call the `printAddDishMessage()` method, which gets all the parameters of the `Dish` object (dishName, dishPrice, dishIngredients) and passes them to the `Ui` to then be printed out to the User.
+
+Separation of Concerns was applied to ensure the `Ui` is only responsible with only displaying messages while the `Menu` deals with the logic of adding dish to the menu. This implementation also encapsulates the details of adding a dish and displaying messages. For example, The `AddDishCommand` class doesn't need to know how the internal details of the dish adding and message printing are performed.
+
+### Adding a Dish
+
+### List Menu
+A `list_menu` command can be used to display all the `Dish` objects stored in `Menu`.
+
+The following class diagram illustrates the relationship between the respective classes involved in the creation and execution of a list_menu command.
+![List Menu Execution](images/class/ListMenuCommandClass.png)
+
+![List Menu Execution](images/sequence/ListMenuCommand_execute.png)
+
+Figure 9: Execution of list_menu command
+
+API: [ListMenuCommand.java]({repoURL}src/main/java/seedu/cafectrl/command/ListMenuCommand.java)
+
+When the `execute()` method of ListMenuCommand is invoked in Main, it checks if the size of the menu by running `menu.getSize()`. 
+
+1) If the menu is empty, it will call its `printEmptyMenu()` method to display to the user a `MENU_EMPTY_MESSAGE` in the Ui object and returns afterward.
+2) If the menu is not empty, it will call its `printFullMenu()` method.
+`printFullMenu()` will first print the top portion of the menu using the Ui object. 
+It then iterates through the `Dish` objects in `Menu` in a "for" loop, using `menu.getDishFromId()` to retrieve the Dish object.
+The `dishName` and `dishPrice` are both access from `Dish` Class using `getName()` and `getPrice()` respectively.
+The data are then packaged nicely in a `leftAlignFormat`, with (indexNum + ". " + dishName," $" + dishPrice) such that
+   e.g. (1. Chicken Rice $2.50) is shown.
+
+### Add Order
+A add_order command can be used to add `order` to an `orderList` in `Sales`.
```

Comment:
You should write "an add_order" instead of "a add_order"
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/49584216/6a9e71ce-5e06-4ffd-8d52-04189d766b14)